discovery-infra/day2.py

```diff
@@ -137,6 +150,63 @@ def set_workers_ips_by_type(tfvars, num_worker_nodes, master_ip_type, worker_ip_
     tfvars[worker_ip_type] = worker_ips_list
 
 
+def wait_nodes_join_ocp_cluster(num_orig_nodes, num_new_nodes):
+    approve_workers_on_ocp_cluster()
+    return get_ocp_cluster_ready_nodes_num() == num_orig_nodes + num_new_nodes
+
+
+def approve_workers_on_ocp_cluster():
+    csrs = get_ocp_cluster_csrs()
+    for csr in csrs:
+        if not csr['status']:
+            csr_name = csr['metadata']['name']
+            ocp_cluster_csr_approve(csr_name)
+
+
+
+def config_etc_hosts(api_vip_ip, api_vip_dnsname):
+    with open("/etc/hosts", "r") as f:
+        hosts_lines = f.readlines()
+    found = False
+    for i, line in enumerate(hosts_lines):
```

Comment:
Use [`for-else`](https://book.pythontips.com/en/latest/for_-_else.html)

```python
    for i, line in enumerate(hosts_lines):
        if api_vip_dnsname in line:
            hosts_lines[i] = api_vip_ip + " " + api_vip_dnsname +"\n"
            break
    else:
        hosts_lines.append(api_vip_ip + " " + api_vip_dnsname +"\n")
```

Reply:
fixed